### PR TITLE
WIP: Defines Log Interceptor

### PR DESCRIPTION
**Pull request by gokulchandra**
Based on #64. Let this be ongoing PR where we can refactor as the design evolves.

- [x] Defines a DSL for adding Interceptors[Initial thoughts]
- [x] Creates Log Interceptor.

### CODE REVIEW

File: src/main/kotlin/io/github/rybalkinsd/kohttp/client/OkHttpClientExt.kt [hunk 1]
```diff
@@ -1,11 +1,22 @@
...
  * @since 0.5.0
  * @author sergey
  */
-fun OkHttpClient.fork(block: ForkClientBuilder.() -> Unit) : OkHttpClient =
-    ClientBuilderImpl(this).apply(block).build()
+fun OkHttpClient.fork(block: ForkClientBuilder.() -> Unit): OkHttpClient =
+        ClientBuilderImpl(this).apply(block).build()
+
+
+class InterceptorsDsl(private val interceptors: MutableList<Interceptor> = mutableListOf()) {
```
Comment (rybalkinsd) [inline src/main/kotlin/io/github/rybalkinsd/kohttp/client/OkHttpClientExt.kt]: need to add test where
`base` client has an interceptor and `fork` client adds one more

File: src/main/kotlin/io/github/rybalkinsd/kohttp/client/ClientBuilder.kt [hunk 2]
```diff
@@ -73,7 +73,7 @@ interface ClientBuilder : ForkClientBuilder {
 
 interface ForkClientBuilder {
     @get:Deprecated(level = DeprecationLevel.ERROR, message = "Write only field")
-    var interceptors: List<Interceptor>
+    var interceptors:  List<Interceptor>
```
[the remark this answers is not in the document]
Comment (gokulchandra) [reply src/main/kotlin/io/github/rybalkinsd/kohttp/client/ClientBuilder.kt]: From what I understand I think it would result in a breaking change.

**Comment by gokulchandra** [comment]
Updated to add a basic LoggerInterceptor. Sample output looks like:

`[2019-01-28T04:17:42.885Z] GET 200 - 1743ms https://postman-echo.com/get`

**Comment by rybalkinsd** [comment]
@gokulchandra I'm planning to merge it during the weekend.
Will also update FormBody.

Hope we'll be able to release 0.8.0 ~ 9 Feb (1 month after 0.7.0)

**Comment by rybalkinsd** [comment]
@gokulchandra Also pls, if ok with your time, check if the "Codacy/PR Quality Review" is false positive or there is a real issue

**Comment by gokulchandra** [comment]
@rybalkinsd I did take a look the @Codacy failure. It isn't able to run on this PR, let me create a new one and see.

**Comment by rybalkinsd** [comment]
looks like an internal problem in codacy. already merged

**Pull request closed without merging**